beacon-chain/sync/service_test.go

```diff
@@ -265,7 +265,22 @@ func TestBlockRequestErrors(t *testing.T) {
 	}
 
 	ss.blockRequestBySlot <- invalidmsg
+	ss.cancel()
+	exitRoutine <- true
```

Comment:
This test had the same issue. They need to be separated, and the `exitRoutine` check was missing.